### PR TITLE
Rework generic KeyStoreOptions

### PR DESCRIPTION
**Pull request by vietj**

### CODE REVIEW

File: src/main/java/io/vertx/core/net/KeyStoreOptionsBase.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+ *
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ * @author <a href="http://tfox.org">Tim Fox</a>
+ */
+class KeyStoreOptionsBase implements KeyCertOptions, TrustOptions {
+
+  private KeyStoreHelper helper;
+  private String type;
+  private String password;
+  private String path;
```
Comment (sberyozkin) [inline src/main/java/io/vertx/core/net/KeyStoreOptionsBase.java:37]: @vietj Hi Julien - it looks good, can this base class include `provider` now ?

Comment (vietj) [reply src/main/java/io/vertx/core/net/KeyStoreOptionsBase.java:37]: yes, but how can we test it ?

Comment (pmlopes) [reply src/main/java/io/vertx/core/net/KeyStoreOptionsBase.java:37]: I think you can use "jce" as provider to load a jceks file. Although this provider is always present if I'm not mistaken.

Comment (vietj) [reply src/main/java/io/vertx/core/net/KeyStoreOptionsBase.java:37]: ok, I'll have a look, perhaps we can write an integration test with a mock provider that delegates to JCE ?

Comment (vietj) [reply src/main/java/io/vertx/core/net/KeyStoreOptionsBase.java:38]: it is actually used `KeyStore.getInstance("JKS").getProvider().getName()` that returns the _SUN_ value

**Comment by vietj** [comment]
I updated the PR with the provider, I think now this is ready for review.

File: src/main/java/io/vertx/core/net/KeyStoreOptions.java [hunk 2]
```diff
@@ -26,33 +27,29 @@
...
- * options.setKeyStore(new KeyStoreOptions().setType("JKS").setPath("/mykeystore.jks").setPassword("foo"));
+ * options.setKeyCertOptions(new KeyStoreOptions().setType("JKS").setPath("/mykeystore.jks").setPassword("foo"));
  * </pre>
  *
  * Or directly provided as a buffer:
  *
  * <pre>
  * Buffer store = vertx.fileSystem().readFileBlocking("/mykeystore.jks");
- * options.setKeyStore(new JKSOptions().setType("JKS").setValue(store).setPassword("foo"));
+ * options.setKeyCertOptions(new JKSOptions().setType("JKS").setValue(store).setPassword("foo"));
```
Comment (pmlopes) [inline src/main/java/io/vertx/core/net/KeyStoreOptions.java:37]: `new KeyStoreOptions()`

**Review by pmlopes — APPROVED** [review]
Just a small typpo on the javadoc

**Pull request merged**